### PR TITLE
Added variable-scope markdown file

### PR DESCRIPTION
**Pull request by walstromowen**
### Description

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by walstromowen** [comment]
Hello! Just a question I tried pushing two different commits from the same branch to the remote repo. The second push gets rejected with the following error.

error: failed to push some refs to 'https://github.com/walstromowen/CodecadeDocs.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

I assume this is because I created a pull request for the first commit and it has not been approved. I tried fetching and merging from the remote repo but errors persist. If you have any insight into this problem to share I would greatly appreciate it!

**Comment by SSwiniarski** [comment]
A new PR should be a separate branch off your main branch. I'd sync the main branch, create a new branch off of it, and copy the changes to that branch.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:1]: This is missing a metadata section, which is a requirement. Refer to the [entry template](https://github.com/Codecademy/docs/blob/main/documentation/entry-template.md) and the [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) to see what's required. You can also look at the source of other JavaScript entries to see what we want.

Also, with the metadata in place, we can remove this header. The title will be in the metadata and that's what will populate the entry.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:5]: There needs to be a line break before and after headers.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:3]:
```suggestion
**Variable scope** refers to the areas where variables are visible and accessible. Before ES6 was introduced in 2015, JavaScript used only "global scope" and "function scope".
```

Generally, we only bold the first occurrence of the subject term.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:6]:
```suggestion
A variable accessible anywhere in a program is said to have global scope and is called a global variable. A global variable must be declared outside all functions in a program.
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:17]: There need to be line breaks before and after code blocks.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:30]: Add a line break after header.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:31]:
```suggestion
A variable declared anywhere within a function is said to have function scope and is called a local variable. Once a local variable is declared, a local variable is said to be local, or accessible, anywhere within the function it was declared in.
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:50]: Add line break after header.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (SSwiniarski) [inline content/javascript/concepts/variable-scope/variable-scope.md:53]:
```suggestion
A variable declared with `let` or `const` within a block { }, is said to have block scope and is also called a local variable. However this local variable is only considered local to the block it was declared in.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@walstromowen I did a first review and left a few comments. Most are just formatting issues, but importantly, you need to include a metadata section in your entry.

**Comment by walstromowen** [comment]
> @walstromowen: A new PR should be a separate branch off your main branch. I'd sync the main branch, create a new branch off of it, and copy the changes to that branch.

Do I need to make a new pull request every time I make a commit? I apologize, this is my first time contributing to an open source project with git. From my understanding, I can... 

1. Use git pull "remote-repo-name" "local-branch-name" to merge changes made by other contributors to my local repo
2. Make changes discussed above on local branch
3. Commit changes on local branch
4. Push committed changes from the local branch to remote repo

The first time I did this, I made a pull request right after step 4 to give editors access to my work. However, I just made a new commit today to the same branch and I am not sure you or other editors can see it?

**Comment by SSwiniarski** [comment]
@walstromowen You shouldn't need to. Once you make a PR on a branch, you should see subsequent commits to that branch.

**Comment by walstromowen** [comment]
Hi! I apologize for all of the redundant commits. I need to remove some commits from this pull request's history as there really is only one important commit. I believe this is achieved using the git rebase command? I tried closing this pull request and then pushed a new commit from the same using the same branch as the deleted request but I think it re-opened this pull request? Unfortunately I am confused and would appreciate any help!

**Comment by SSwiniarski** [comment]
@walstromowen I wouldn't worry about the number of commits as long as the PR is just changing what you want to change.

**Review by SSwiniarski — APPROVED** [review]
@walstromowen It looks like this is ready for the next review.

**Comment by walstromowen** [comment]
Thank you! Is there anything I need to do on my side or do I just need to wait for reviews to finish?

**Comment by SSwiniarski** [comment]
@walstromowen nothing you need to do until another maintainer comes and reviews this PR.

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:36]: For code snippets in docs entries, use two spaces instead of the standard 4.
```suggestion
  a = 1; //global variable
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:50]:
```suggestion
  var a = 1;
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:51]:
```suggestion
  if(a === 1){
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:52]:
```suggestion
    var b = 2;
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:53]: Capitalize the first letter of the comments in doc entry snippets
```suggestion
    alert(a); // Code here can use variable a
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:54]:
```suggestion
    alert(b); // Code here can use variable b
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:56]:
```suggestion
  alert(a); // Code here can also use variable a
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:57]:
```suggestion
  alert(b); // Code here can also use variable b
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:62]:
```suggestion
alert(a); // Code here CANNOT use variable a
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:73]:
```suggestion
  let a = 1;
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:74]:
```suggestion
  if(a === 1){
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:75]:
```suggestion
    let b = 2;
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:76]:
```suggestion
    alert(a); // Code here can use variable a
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:77]:
```suggestion
    alert(b); // Code here can use variable b
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:79]:
```suggestion
  alert(a); // Code here can also use variable a
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:80]:
```suggestion
  alert(b); // Code here CANNOT use variable b
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:85]:
```suggestion
alert(a); // Code here CANNOT use variable a
```

File: content/javascript/concepts/variable-scope-js/variable-scope-js.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope-js/variable-scope-js.md:86]:
```suggestion
alert(b); // Code here also CANNOT use variable b
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@walstromowen I've added a second review that includes a title change. Please also change the file name so that the file path is `content/javascript/concepts/variable-scope/variable-scope.md` Let me know if you have any questions.

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope/variable-scope.md:55]:
```suggestion
  }
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope/variable-scope.md:73]:
```suggestion
  let a = 1;
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope/variable-scope.md:74]:
```suggestion
  if(a === 1){
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope/variable-scope.md:75]:
```suggestion
    let b = 2;
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope/variable-scope.md:76]:
```suggestion
    alert(a); // Code here can use variable a
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope/variable-scope.md:77]:
```suggestion
    alert(b); // Code here can use variable b
```

File: content/javascript/concepts/variable-scope/variable-scope.md
Comment (KTom101) [inline content/javascript/concepts/variable-scope/variable-scope.md:78]:
```suggestion
  }
```

**Review by KTom101 — COMMENTED** [review]
@walstromowen I added more changes. Mainly regarding the spacing in the last two examples.

**Comment by walstromowen** [comment]
@KTom101 thank you for your review. I think I fixed it this time.

**Review by SSwiniarski — APPROVED** [review]
@KTom101, @walstromowen Looks good to me.

**Comment by walstromowen** [comment]
@SSwiniarski and KTom101 Thank you! let me know if there is any other changes I should make.

**Review by KTom101 — COMMENTED** [review]
@walstrmowen I've added a few more changes to this.

**Comment by KTom101** [comment]
@walstromowen How are the changes coming along? Do you need any help with this?

**Comment by walstromowen** [comment]
@ KTom101 I have made the changes as requested. Thank you

**Review by KTom101 — APPROVED** [review]
@walstromowen LGTM 👍

**Comment by KTom101** [comment]
@walstromowen Do you need any assistance fixing the failing checks?

**Comment by walstromowen** [comment]
@ KTom101 I would appreciate that! thanks!

**Comment by KTom101** [comment]
@walstromowen I've fixed the checks. So you know I ran `npm run format:verify` and `npm run format:all` to do this.

**Comment by walstromowen** [comment]
Thank You!Do you know how I can get my profile to appear as a contributor on the docs page?   Owen Walstrom From: KTom101Sent: Friday, February 3, 2023 5:26 PMTo: Codecademy/docsCc: walstromowen; MentionSubject: Re: [Codecademy/docs] Added variable-scope markdown file (PR ***@***.*** I've fixed the checks. So you know I ran npm run format:verify and npm run format:all to do this.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by KTom101** [comment]
@walstromowen You can add your GitHub username to your Codecademy profile by doing the following:

1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:

YourGitHubName

2.)	Then log out and log back into Codecademy using your GitHub login:

ReLogIn

Let me know if you need more detailed instructions! 😁

**Comment by KTom101** [comment]
@walstromowen Also, feel free to take a look at the [issues tab](https://github.com/Codecademy/docs/issues) to see if there are any recently added issues you want to work on or let us know if there is something not in Docs that you wish to add.

**Comment by walstromowen** [comment]
Thank you!  Appreciate all of your help! Sincerely, Owen Walstrom From: KTom101Sent: Friday, February 3, 2023 5:48 PMTo: Codecademy/docsCc: walstromowen; MentionSubject: Re: [Codecademy/docs] Added variable-scope markdown file (PR ***@***.*** Also, feel free to take a look at the issues tab to see if there are any recently added issues you want to work on or let us know if there is something not in Docs that you wish to add.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Pull request merged**
